crates/driver/src/infra/api/routes/solve/dto/solved.rs

```diff
@@ -38,9 +38,17 @@ impl Solution {
                 .map(|(order_id, amounts)| {
                     (
                         order_id.into(),
-                        TradedAmounts {
-                            sell_amount: amounts.sell.into(),
-                            buy_amount: amounts.buy.into(),
+                        TradedOrder {
+                            side: match amounts.side {
```

Comment:
As a somewhat related comment here, I generally I think implementing the `From` trait is not always great because it makes it harder to find the code that actually performs the conversion in an IDE. Unless we use it in something that requires `Into` I'd always prefer to use a dedicated from/into method rather than implementing the trait. But your mileage may vary.

Reply:
will stick with no trait implementation in this case